From fe713d8a30e270d536c000b5a4b083836c0a2a56 Mon Sep 17 00:00:00 2001
Message-ID: <fe713d8a30e270d536c000b5a4b083836c0a2a56.1721514687.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1721514686.git.sharlatanus@gmail.com>
References: <cover.1721514686.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 15 Jul 2024 20:15:27 +0100
Subject: [PATCH 31/34] gnu: Add python-mpl-scatter-density.

* gnu/packages/python-science.scm (python-mpl-scatter-density): New variable.

Change-Id: Id4ea80b2c62642affbe1944e0c5126a1c53d16c3
---
 gnu/packages/python-science.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/python-science.scm b/gnu/packages/python-science.scm
index 7012a942d3..4bb8ac342d 100644
--- a/gnu/packages/python-science.scm
+++ b/gnu/packages/python-science.scm
@@ -779,6 +779,37 @@ (define-public python-meshzoo
 spheres, cubes, etc.")
     (license license:gpl3+)))
 
+(define-public python-mpl-scatter-density
+  (package
+    (name "python-mpl-scatter-density")
+    (version "0.7")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "mpl-scatter-density" version))
+       (sha256
+        (base32 "083n4xkwjmxvij9i1xhfnxz8vk39ay0baam4nf0khvcihw47bkna"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      ;; See <https://github.com/astrofrog/mpl-scatter-density/issues/42>.
+      #:test-flags #~(list "-k" "not test_default_dpi")))
+    (propagated-inputs
+     (list python-fast-histogram
+           python-matplotlib
+           python-numpy))
+    (native-inputs
+     (list python-pytest
+           python-pytest-cov
+           python-pytest-mpl
+           python-setuptools-scm))
+    (home-page "https://github.com/astrofrog/mpl-scatter-density")
+    (synopsis "Matplotlib helpers to make density scatter plots")
+    (description
+     "This package provides functionality to make it easy to make scatter
+density maps, both for interactive and non-interactive use.")
+    (license license:bsd-2)))
+
 (define-public python-pyamg
   (package
     (name "python-pyamg")
-- 
2.41.0

